### PR TITLE
feat: fiber support filter tag

### PR DESCRIPTION
**Pull request by devhaozi**
Closes goravel/goravel#304
Closes goravel/goravel#259
Closes goravel/goravel#247
Closes goravel/goravel#313

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: context_request_test.go [hunk 1]
```diff
@@ -1360,6 +1360,41 @@ func TestRequest(t *testing.T) {
 			expectCode: http.StatusBadRequest,
 			expectBody: "Validate fail: map[name:map[required:name is required to not be empty]]",
 		},
+		{
+			name:   "POST with validator use filter and validate request pass",
+			method: "POST",
+			url:    "/validator/validate-request/success",
```
Comment (hwbrzzl) [inline context_request_test.go:1366]:
```suggestion
			url:    "/validator/filter/success",
```

File: context_request_test.go [hunk 2]
```diff
@@ -1360,6 +1360,41 @@ func TestRequest(t *testing.T) {
...
+						return ctx.Response().String(http.StatusBadRequest, fmt.Sprintf("Validate fail: %+v", validateErrors.All()))
+					}
+
+					return ctx.Response().Success().Json(contractshttp.Json{
+						"name": createUser.Name,
+					})
+				})
+
+				payload := strings.NewReader(`{
+					"name": " Goravel"
```
Comment (hwbrzzl) [inline context_request_test.go:1388]:
```suggestion
					"name": " Goravel "
```

File: context_request_test.go [hunk 3]
```diff
@@ -1360,6 +1360,41 @@ func TestRequest(t *testing.T) {
 			expectCode: http.StatusBadRequest,
 			expectBody: "Validate fail: map[name:map[required:name is required to not be empty]]",
 		},
+		{
```
Comment (hwbrzzl) [inline context_request_test.go:1363]: Q: Does `Get` work?

**Review by hwbrzzl — COMMENTED** [review]
Great work! could you please also add this feature for Gin?

File: context_request_test.go (hunk 2 above)
Comment (hwbrzzl) [reply context_request_test.go:1423]: Yes, we need to verify the `PrepareForValidation` is called first, then `Filter`

Comment (hwbrzzl) [reply context_request_test.go:1423]: So we can add a space to the end of ` Goravel`, and check `Goravel 1`.

**Comment by devhaozi** [comment]
> Great work! could you please also add this feature for Gin?

Added

File: context.go [hunk 4]
```diff
@@ -48,6 +48,10 @@ func (c *Context) Context() context.Context {
 	return c.instance.UserContext()
 }
 
+func (c *Context) SetContext(ctx http.Context) {
+	c.instance.SetUserContext(ctx)
```
Comment (hwbrzzl) [inline context.go:52]: Q: Will cover the value that is set by `WithValue`?

File: go.mod [hunk 5]
```diff
@@ -3,77 +3,64 @@ module github.com/goravel/fiber
 go 1.20
 
 require (
-	github.com/gofiber/fiber/v2 v2.51.0
+	github.com/gofiber/fiber/v2 v2.50.0
```
Comment (hwbrzzl) [inline go.mod:6]: Downgrade?

File: context.go (hunk 4 above)
Comment (hwbrzzl) [reply context.go:52]: We can remove this method for now.

File: context_request.go [hunk 6]
```diff
@@ -54,6 +55,11 @@ func (r *ContextRequest) All() map[string]any {
 	data := make(map[string]any)
 
 	var mu sync.RWMutex
+	for k, v := range r.instance.AllParams() {
+		mu.Lock()
+		data[k] = v
+		mu.Unlock()
+	}
```
Comment (hwbrzzl) [inline context_request.go:62]: Good catch!

**Review by hwbrzzl — APPROVED** [review]
LGTM, could you please list all supported filter rules in the PR description?

**Pull request merged**